tests/DCPS/Compiler/xcdr/xcdr.cpp

```diff
@@ -130,6 +198,19 @@ ::testing::AssertionResult assert_DataView(
     << result.str();
 }
 
+template<typename T>
+void deserialize_compare(
+  const Encoding& encoding, const unsigned char data[], const size_t size,
+  const T& expected, const size_t message_block_size = 1024)
+{
+  ACE_Message_Block mb(message_block_size);
+  mb.copy((const char*)data, size);
```

Comment:
```suggestion
  const Encoding& encoding, const DataView& data, const T& expected)
{
  ACE_Message_Block mb(data.size);
  mb.copy((const char*)data.data, data.size);
```

Make use of `DataView` and go ahead and make it the `ACE_Measage_Block` size.

Reply:
This differs from the rest of the tests which use a bigger buffer (1024) to test deserialization. I asked before about the previous design thought. Anyway, I like this better. It took more to preserve the previous approach.

Reply:
I realized in this case we know the exact amount of data going into `ACE_Message_Block` because it's just a copy. The 1024 size in the other function is for handling the possibility of our serialization functions writing more than what they should have.